### PR TITLE
#1690 Update independent assessments UI

### PR DESCRIPTION
**Pull request by HalcyonJAC**
## What's included?
https://app.zenhub.com/files/201935567/942a1707-30b2-4c1e-b331-8dd7ee8670a6/download

Note: this PR is related to:
- Apply site: [#912 Select Assessor Type](https://github.com/jac-uk/apply/pull/911)
- Digital platform: [Update assessments functions](https://github.com/jac-uk/digital-platform/pull/761)
- Assessment site: [Decline assessment](https://github.com/jac-uk/assessments/pull/153)

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
1. Go to `Independent Assessments` of an exercise.
2. Click `Start Assessments` to start assessments.

    Not requested:
    - Check `Assessor type` column in the table.
    - Select assessments and click `Test requests` to send test assessment requests.
    - Select assessments and click `Send requests` to send assessment requests. And then the selected assessments should appear in `Requested` tab and `Sent` statistical figure should increase one.
    - Select assessments and click `Delete` and then the selected assessments should move to `Deleted` tab.

    https://user-images.githubusercontent.com/79906532/188426956-98d3a24f-563c-4c9f-9a70-7d8d9449fc2d.mov

    Requested:
    - Check `Assessor type` column in the table.
    - Select assessments and click `Test reminders` to send test assessment reminders.
    - Select assessments and click `Send reminders` to send assessment reminders.
    - Select assessments and click `Cancel`. It should be able to type in some reason. And then the selected assessments should move to `Cancelled` tab.
    - Select assessments and click `Reset`. And then the selected assessments should move to `Not requested` tab and `Sent` statistical figure should decrease one.
    - Select assessments and click `Delete` and then the selected assessments should move to `Deleted` tab.

    https://user-images.githubusercontent.com/79906532/188428271-8480e72b-8da4-43ce-8501-122f83ab4543.mov

    Completed:
    - Check `Assessor type` column in the table.
    - Select assessments and click `Cancel`. It should be able to type in some reason. And then the selected assessments should move to `Cancelled` tab and `Completed` statistical figure should decrease one..
    - Select assessments and click `Reset`. And then the selected assessments should move to `Not requested` tab and `Sent` and `Completed` statistical figures should decrease one.
    - Select assessments and click `Delete`. And then the selected assessments should move to `Requested` tab and `Completed` statistical figure should decrease one.
 
    Cancelled & Decline:
    - Check `Assessor type` column in the table.
    - Check the `Reason` column in the table.
    - Select assessments and click `Reset`. And then the selected assessments should move to `Not requested` tab and `Sent` statistical figure should decrease one.
 
    https://user-images.githubusercontent.com/79906532/188428499-03484677-8295-4352-88bc-2b9936d5dba4.mov

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Additional context
Include screen grabs, video demo, notes etc.

## Related permissions
- No permission changes required

---
PREVIEW:DEVELOP

### CODE REVIEW

File: src/views/Exercise/Tasks/IndependentAssessments.vue [hunk 1]
```diff
@@ -72,186 +72,530 @@
...
+          <Table
+            key="notrequested"
+            data-key="id"
+            :data="assessmentsNotRequestedRecords"
+            :page-size="50"
+            :columns="tableColumns"
+            multi-select
+            :selection.sync="selectedItems"
+            :custom-search="{
+              placeholder: 'Search candidate names',
```
Comment (warrensearle) [inline src/views/Exercise/Tasks/IndependentAssessments.vue:123]: There are lots of instances of the `Table` component whilst only one shows at a time. As the contents are so similar can we consider using just one instance? Right now if we wish to change something, for example the custom-search plugin, then we will need to change it in 5 places.

**Review by warrensearle — CHANGES_REQUESTED** [review]
Looking good, just a question about simplifying the code to make it more maintainable which we can discuss if you like.

**Review by warrensearle — APPROVED** [review]
LGTM 👍

**Review by tomlovesgithub — APPROVED** [review]

**Review by andrew-isaac — CHANGES_REQUESTED** [review]
I sent out Adam Schinner IAs on this exercise - https://jac-admin-develop--pr1696-feature-1690-update-seca6n3n.web.app/exercise/RykU5zMVsXd3cwa5yqqQ/tasks/independent-assessments#requested  He moved to requested and I then cancelled it to move him back to not requested. I sent out the IA again but he has now disappeared from all lists and I can't locate him. I did this again with another candidate and it seemed to be ok?

Also when moving between tabs I am having to refresh to get the data to update - It did warm up after time so may be it is just where the data was populating?

**Comment by warrensearle** [comment]
UT results: https://drie.slack.com/archives/C01R6GAAETG/p1665587137605259?thread_ts=1665580898.414769&cid=C01R6GAAETG

**Pull request merged**